packages/govuk-frontend/src/govuk/errors/index.mjs

```diff
@@ -40,3 +40,10 @@ export class SupportError extends GOVUKFrontendError {
     super('GOV.UK Frontend is not supported in this browser')
   }
 }
+
+/**
+ * Indicates that a component has received an illegal configuration
+ */
+export class IllegalConfigurationError extends GOVUKFrontendError {
```

Comment:
Similar to how `TypeError` isn't `IllegalTypeError`, should we flip out the negative?

Some ideas:

```
ConfigError
ConfigValidationError
```

Probably use "Config" versus "Configuration" too to keep it consistent?